bundles/org.openhab.binding.meater/src/main/java/org/openhab/binding/meater/internal/handler/MeaterBridgeHandler.java

```diff
@@ -0,0 +1,169 @@
+
+/**
+ * The {@link MeaterBridgeHandler} is responsible for handling commands, which are
+ * sent to one of the channels.
+ *
+ * @author Jan Gustafsson - Initial contribution
+ */
+@NonNullByDefault
+public class MeaterBridgeHandler extends BaseBridgeHandler {
+
+    private final Logger logger = LoggerFactory.getLogger(MeaterBridgeHandler.class);
+
+    public static final Set<ThingTypeUID> SUPPORTED_THING_TYPES = Collections.singleton(THING_TYPE_BRIDGE);
+
+    private int refreshTimeInSeconds = 300;
+
+    private final Gson gson;
+    private final HttpClient httpClient;
+    private final TranslationProvider i18nProvider;
+    private LocaleProvider localeProvider;
+    private final Map<String, MeaterProbeDTO.Device> meaterProbeThings = new ConcurrentHashMap<>();
+
+    private @Nullable MeaterRestAPI api;
+    private @Nullable ScheduledFuture<?> refreshJob;
+
+    public MeaterBridgeHandler(Bridge bridge, HttpClient httpClient, Gson gson, TranslationProvider i18nProvider,
+            LocaleProvider localeProvider) {
+        super(bridge);
+        this.httpClient = httpClient;
+        this.gson = gson;
+        this.i18nProvider = i18nProvider;
+        this.localeProvider = localeProvider;
+    }
+
+    @Override
+    public void initialize() {
+        MeaterBridgeConfiguration config = getConfigAs(MeaterBridgeConfiguration.class);
+
+        api = new MeaterRestAPI(config, gson, httpClient, localeProvider);
+        refreshTimeInSeconds = config.refresh;
+
+        if (config.email == null || config.password == null) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR,
+                    "@text/config.missing-username-password.description");
+        } else if (refreshTimeInSeconds < 0) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR,
+                    "@text/config.negative-refresh-interval.description");
+        } else {
+            scheduler.execute(() -> {
+                updateStatus(ThingStatus.UNKNOWN);
+                startAutomaticRefresh();
+            });
+        }
+    }
+
+    public Map<String, MeaterProbeDTO.Device> getMeaterThings() {
+        return meaterProbeThings;
+    }
+
+    @Override
+    public Collection<Class<? extends ThingHandlerService>> getServices() {
+        return Collections.singleton(MeaterDiscoveryService.class);
+    }
+
+    @Override
+    public void dispose() {
+        stopAutomaticRefresh();
+    }
+
+    private boolean refreshAndUpdateStatus() {
+        MeaterRestAPI localAPI = api;
+        if (localAPI != null) {
+            if (localAPI.refresh(meaterProbeThings)) {
+                getThing().getThings().stream().forEach(thing -> {
+                    MeaterHandler handler = (MeaterHandler) thing.getHandler();
+                    if (handler != null) {
+                        handler.update();
+                    }
+                });
+                updateStatus(ThingStatus.ONLINE);
```

Comment:
Considering another of my comments, it is needed that you set the bridge status to ONLINE before going through all its childs.

Reply:
Moved to row 128.